src/containers/Network/UpgradeStatus.tsx

```diff
@@ -22,63 +22,108 @@ import {
 } from '../shared/vhsTypes'
 import NetworkContext from '../shared/NetworkContext'
 import { ledgerCompare } from './Nodes'
+import { Loader } from '../shared/components/Loader'
 
-interface DataAggregation {
-  label: string
-  validatorsPercent: number
-  validatorsCount: number
+interface NodeStats {
   nodesPercent: number
   nodesCount: number
 }
 
-export const aggregateData = (
-  validators: ValidatorResponse[],
-  nodes: NodeResponse[],
-): DataAggregation[] => {
-  if (!validators) {
-    return []
-  }
+interface ValidatorStats {
+  validatorsPercent: number
+  validatorsCount: number
+}
 
-  let totalVals = 0
-  let totalNodes = 0
-  interface aggregationTypes {
-    validatorsCount: number
-    nodesCount: number
-  }
+interface ValidatorsAggregation {
+  [label: string]: ValidatorStats
+}
+
+interface NodesAggregation {
```

Comment:
Easier to read when singular
```suggestion
interface NodeAggregation {
```